packages/console/src/pages/ConnectorDetails/index.tsx

```diff
@@ -51,12 +51,9 @@ function ConnectorDetails() {
   const [isReadMeOpen, setIsReadMeOpen] = useState(false);
   const [isSetupOpen, setIsSetupOpen] = useState(false);
   const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
-  /**
-   * TODO: Can add `keepPreviousData` option to useSWR to avoid the flash on `Skeleton`
-   * component when manually trigger `mutate` function. This change is available in `swr@v2.0` or above.
-   */
   const { data, error, mutate } = useSWR<ConnectorResponse, RequestError>(
-    connectorId && `api/connectors/${connectorId}`
+    connectorId && `api/connectors/${connectorId}`,
+    { keepPreviousData: true }
```

Comment:
According to the doc, `keepPreviousData` option is for the case that `key` may change?

Reply:
It simply maintains the previous data, once the new data is loaded and validated then the previous data will be replaced.